internal/util/cephcmds.go

```diff
@@ -65,6 +66,54 @@ func ExecCommand(ctx context.Context, program string, args ...string) (string, s
 	return stdout, stderr, nil
 }
 
+// ExecCommandWithTimeout executes passed in program with args, timeout and
+// returns separate stdout and stderr streams. If the command is not executed
+// within given timeout, the process will be killed. In case ctx is not set to
+// context.TODO(), the command will be logged after it was executed.
+func ExecCommandWithTimeout(
+	ctx context.Context,
+	timeout time.Duration,
+	program string,
+	args ...string) (
+	string,
+	string,
+	error) {
+	var (
+		sanitizedArgs = StripSecretInArgs(args)
+		stdoutBuf     bytes.Buffer
+		stderrBuf     bytes.Buffer
+	)
+
+	cctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	cmd := exec.CommandContext(cctx, program, args...) // #nosec:G204, commands executing not vulnerable.
+	cmd.Stdout = &stdoutBuf
+	cmd.Stderr = &stderrBuf
+
+	err := cmd.Run()
+	stdout := stdoutBuf.String()
+	stderr := stderrBuf.String()
+	if err != nil {
+		err = fmt.Errorf("an error (%w) and stderror (%s) occurred while running %s args: %v",
+			err,
+			stderr,
+			program,
+			sanitizedArgs)
```

Comment:
Since this a generic utility helper, it would be nice to differentiate between the cases of a timeout and a command failing of its own accord within a timeout a bit more explicitly, so that instead of "signal: killed" or something like that the error message actually mentions the timeout.  Perhaps check for `context.DeadlineExceeded`?

Reply:
checking is it possible with it.

Reply:
update code to have `timeout: context deadline exceeded` error in the error message.